### PR TITLE
Feature/mlir op split init

### PR DESCRIPTION
**Pull request by scott-xmos**
prototype op split: only splits in 2, only matches on conv, only handles dimensions divisible by 2

### CODE REVIEW

File: experimental/xformer/Transforms/OpSplit.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 XMOS LIMITED. This Software is subject to the terms of the
+// XMOS Public License: Version 1
+
+#include "IR/XCoreOps.h"
```
Comment (ghost) [inline experimental/xformer/Transforms/OpSplit.cpp]: Please remove unused headers

File: experimental/xformer/Transforms/OpSplit.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  }
+  StringRef getArgument() const final { return "xcore-opsplit"; }
+  StringRef getDescription() const final { return "OpSplit."; }
+  void runOnOperation() override;
+};
+
+struct OpSplitPattern : public OpRewritePattern<TFL::Conv2DOp> {
+  using OpRewritePattern<TFL::Conv2DOp>::OpRewritePattern;
+
+  // LogicalResult matchAndRewrite(DummyStridedSliceOp stridedSliceOriginal,
```
Comment (ghost) [inline experimental/xformer/Transforms/OpSplit.cpp]: Remove the commented out line please

File: experimental/xformer/Transforms/OpSplit.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  LogicalResult matchAndRewrite(TFL::Conv2DOp convOriginal,
+                                PatternRewriter &rewriter) const override {
+
+    static constexpr char kSplitLabel[] = "__split_op__";
+
+    // Do not split ops already split
+    if (convOriginal->hasAttr(kSplitLabel))
+      return failure();
+
```
Comment (ghost) [inline experimental/xformer/Transforms/OpSplit.cpp:35]: We should fail to match on restrictions we have. For e.g, if we cannot support conv2d with odd dimensions, should return failure(). We can slowly remove the failures as we support more cases.

Comment (ghost) [inline experimental/xformer/Transforms/OpSplit.cpp:35]: We should fail to match on restrictions we have. For e.g, if we cannot support conv2d with odd dimensions, should return failure(). We can slowly remove the failures as we support more cases.

File: experimental/xformer/Transforms/Passes.cpp [hunk 4]
```diff
@@ -22,6 +22,7 @@ void buildXCorePassPipeline(OpPassManager &pm) {
   pm.addPass(createPad3to4Conv2DPass());
   pm.addPass(createApplyTFLPatternsPass());
   // XC passes
+  pm.addPass(createOpSplitPass());
```
Comment (ghost) [inline experimental/xformer/Transforms/Passes.cpp]: Please move the pass to the TFL passes list

Comment (ghost) [inline experimental/xformer/Transforms/Passes.cpp]: At least for now, as we develop it further, we should gate the opsplit pass in a command-line option, something like `--xcore-opsplit`.

Comment (scott-xmos) [reply experimental/xformer/Transforms/Passes.cpp]: changed pass name to --xcore-op-split so op split is 2 words, made command-line option --xcore-op-split-tensor-arena so its name is different than passes name

**Review by ghost — APPROVED** [review]
+1

**Pull request merged**